src/pods/edit-table/components/nested-field-grid.tsx

```diff
@@ -44,6 +44,26 @@ export const NestedFieldGrid: React.FC<NestedFieldGridProps> = ({
     collapsed: { opacity: 0, height: 0 },
   };
 
+  const nameInputRefRecord = React.useRef<Record<string, HTMLInputElement>>({});
+
+  const [newFielId, setNewFieldId] = React.useState<GUID>('');
+
+  const handleAddField = (fieldId: GUID, isChildren: boolean) => {
+    const newFieldId = GenerateGUID();
+    setNewFieldId(newFieldId);
+    onAddField(fieldId, isChildren, newFieldId);
+  };
+
+  React.useEffect(() => {
+    const input = nameInputRefRecord.current[newFielId];
+
+    if (input) {
+      input.focus();
+      input.select();
+      setNewFieldId('');
+    }
+  }, [newFielId, nameInputRefRecord.current]);
```

Comment:
This can be encapsulated in a customHook

```tsx
const useInputRefFocus = (onAddField : (fieldId, isChildren, newFieldId) => void) => {
  const nameInputRefRecord = React.useRef<Record<string, HTMLInputElement>>({});
  const [newFielId, setNewFieldId] = React.useState<GUID>('');

  const handleAddField = (fieldId: GUID, isChildren: boolean) => {
    const newFieldId = GenerateGUID();
    setNewFieldId(newFieldId);
    onAddField(fieldId, isChildren, newFieldId);
  };

  React.useEffect(() => {
    const input = nameInputRefRecord.current[newFielId];

    if (input) {
      input.focus();
      input.select();
      setNewFieldId('');
    }
  }, [newFielId, nameInputRefRecord.current]);

  return {nameInputRefRecord, handleAddField}
}
```


Then usage:

```diff
- const nameInputRefRecord = React.useRef<Record<string, HTMLInputElement>>({});
-
-  const [newFielId, setNewFieldId] = React.useState<GUID>('');
-
-  const handleAddField = (fieldId: GUID, isChildren: boolean) => {
-    const newFieldId = GenerateGUID();
-    setNewFieldId(newFieldId);
-    onAddField(fieldId, isChildren, newFieldId);
-  };
-
-  React.useEffect(() => {
-    const input = nameInputRefRecord.current[newFielId];
-
-    if (input) {
-      input.focus();
-      input.select();
-      setNewFieldId('');
-    }
-  }, [newField, nameInputRefRecord.current]);

+ const {nameInputRefRecord, handleAddField} = useInputRefFocus(onAddField);
```